pkg/apis/v1alpha1/vpc_types.go

```diff
@@ -0,0 +1,138 @@
+/* Copyright � 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+   package v1alpha1
+
+   import (
+	   corev1 "k8s.io/api/core/v1"
+	   metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+   )
+   
+   type VPCStatusCondition string
+   
+   const (
+	   VPCReady    VPCStatusCondition = "Ready"
+	   VPCNotReady VPCStatusCondition = "NotReady"
+   )
+   
+   //+kubebuilder:object:root=true
+   //+kubebuilder:subresource:status
+   //+kubebuilder:storageversion
+   
+   // VPC is the Schema for the VPC API
+   type VPC struct {
+	   metav1.TypeMeta   `json:",inline"`
+	   metav1.ObjectMeta `json:"metadata,omitempty"`
+   
+	   Spec   VPCSpec   `json:"spec,omitempty"`
+	   Status VPCStatus `json:"status,omitempty"`
+   }
+   
+   //+kubebuilder:object:root=true
+   
+   // VPCList contains a list of VPC
+   type VPCList struct {
+	   metav1.TypeMeta `json:",inline"`
+	   metav1.ListMeta `json:"metadata,omitempty"`
+	   Items           []VPC `json:"items"`
+   }
+   
+   // VPCSpec defines VPC configuration
+   type VPCSpec struct {
+	   // The edge cluster on which the networking elements for the Org will be created.
```

Comment:
cluster -> clusters

for the Org -> for the VPC?

Reply:
@jianjuns  Do you think we need to keep the same structure with NSX VPC API as below with site_infos which include edge cluster paths or just use edge cluster path? maxItem:1 for site_infos.
                   "site_infos": [
                    {
                        "edge_cluster_paths": [
                            "/infra/sites/default/enforcement-points/default/edge-clusters/4a033a2f-d42e-421d-a41b-33333656278a"
                        ],
                        "site_path": "/infra/sites/default"
                    }
                ]

Reply:
No, no need to include all fields in NSX-T API. Just add what are necessary for now.

Reply:
As we talked, we need not to include fields which are not necessary. These parameters are already included in VPCNetworkConfiguration, so we can remove them from VPCSpec, unless keeping them can simplify the NSX-T Operator implementation.